src/actix/api/snapshot_api.rs

```diff
@@ -198,29 +211,34 @@ async fn get_full_snapshot(
     path: web::Path<String>,
 ) -> impl Responder {
     let snapshot_name = path.into_inner();
-    do_get_full_snapshot(toc.get_ref(), &snapshot_name).await
+    do_get_full_snapshot(&toc, &snapshot_name).await
 }
 
 #[delete("/snapshots/{snapshot_name}")]
 async fn delete_full_snapshot(
-    toc: web::Data<TableOfContent>,
+    dispatcher: web::Data<Dispatcher>,
     path: web::Path<String>,
+    params: Query<SnapshottingParam>,
 ) -> impl Responder {
     let snapshot_name = path.into_inner();
     let timing = Instant::now();
-    let response = do_delete_full_snapshot(toc.get_ref(), &snapshot_name).await;
+    let wait = params.wait.unwrap_or(true);
+    let response = do_delete_full_snapshot(dispatcher.get_ref(), &snapshot_name, wait).await;
```

Comment:
Is it also possible to return 202 for the delete operations as well?
This would make the API more consistent.